examples/browser-to-server/tests/test.js

```diff
@@ -0,0 +1,96 @@
+/* eslint-disable no-console */
+import { test, expect } from '@playwright/test'
+import { playwright } from 'test-util-ipfs-example'
+import { spawn, exec } from 'child_process'
+import { existsSync } from 'fs'
+
+// Setup
+const play = test.extend({
+  ...playwright.servers()
+})
+
+async function spawnGoLibp2p() {
+  if (!existsSync('../../examples/go-libp2p-server/go-libp2p-server')) {
+    await new Promise((resolve, reject) => {
+      exec('go build',
```

Comment:
same here, might as well just always run it, no? instead of building it first?

Reply:
Good question! this should be documented. `go run ./main.go` builds the program then starts a subprocess for the program. SIGINT on the parent process doesn't stop the subprocess (try running `go run` in the background and issuing a `kill -2 pid`). By running the binary directly we avoid the indirection and SIGINT works as expected.

I wonder if we could SIGKILL this instead and it would work?

I don't think it's worth changing in this PR.

Reply:
I agree that it is probably out of scope. More of a note.